allennlp/training/optimizers.py

```diff
@@ -26,13 +27,50 @@ class Optimizer(Registrable):
     default_implementation = "adam"
 
     @classmethod
-    def from_params(cls, model_parameters: List[torch.nn.Parameter], params: Params):
+    def from_params(cls, model_parameters: List, params: Params):
         if isinstance(params, str):
             optimizer = params
             params = Params({})
         else:
             optimizer = params.pop_choice("type", Optimizer.list_available())
-        return Optimizer.by_name(optimizer)(model_parameters, **params.as_dict()) # type: ignore
+
+        # make the parameter groups if need
+        groups = params.pop("parameter_groups", None)
+        if groups:
+            # input to optimizer is list of dict
+            # each dict contains {'params': [list of parameters], 'lr': 1e-3, ...}
+            # see: http://pytorch.org/docs/0.3.0/optim.html?#per-parameter-options
+            #
+            # groups contains something like:
+            #"parameter_groups": [
+            #       [['regex1', 'regex2'], {'lr': 1e-3},
+            #        ['regex3'], {'lr': 1e-4}]
+            #]
+            # last entry is for the parameters not in any regex
```

Comment:
How are config options passed for the default group?

Reply:
Oh I see they are just from the top level config. Can you mention that in a comment somewhere?